mlflow/sagemaker/__init__.py

```diff
@@ -1747,6 +1776,238 @@ def _does_model_exist(model_name, sage_client):
         return True if response else False
 
 
+class SageMakerDeploymentClient(BaseDeploymentClient):
+    """
+    Initialize a deployment client for SageMaker. The default region and assumed role ARN will
+    be set according to the value of the `target_uri`.
+
+    :param target_uri: A URI that follows one of the following formats:
+
+                       - ``sagemaker``: This will set the default region to `us-west-2` and
+                         the default assumed role ARN to `None`.
+
+                       - ``sagemaker:/region_name``: This will set the default region to
+                         `region_name` and the default assumed role ARN to `None`.
+
+                       - ``sagemaker:/region_name/assumed_role_arn``: This will set the default
+                         region to `region_name` and the default assumed role ARN to
+                         `assumed_role_arn`.
+
+                       When an `assumed_role_arn` is provided without a `region_name`,
+                       an MlflowException will be raised.
+    """
+
+    def __init__(self, target_uri):
+        super(SageMakerDeploymentClient, self).__init__(target_uri=target_uri)
+
+        # Default region_name and assumed_role_arn when
+        # the target_uri is `sagemaker` or `sagemaker:/`
+        self.region_name = DEFAULT_REGION_NAME
+        self.assumed_role_arn = None
+        self._get_values_from_target_uri()
+
+    def _get_values_from_target_uri(self):
+        parsed = urllib.parse.urlparse(self.target_uri)
+        values_str = parsed.path.strip("/")
+
+        if not parsed.scheme or not values_str:
+            return
+
+        separator_index = values_str.find("/")
+        if separator_index == -1:
+            # values_str would look like us-east-1
+            self.region_name = values_str
+        else:
+            # values_str would look like us-east-1/arn:aws:1234:role/assumed_role
+            self.region_name = values_str[:separator_index]
+            self.assumed_role_arn = values_str[separator_index + 1 :]
```

Comment:
Can we handle the case where there may be multiple interior slashes in a path? e.g. `sagemaker:/us-west-2////arn:iam:my_role`?